sentinel-adapter/sentinel-spring-webmvc-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/spring/webmvc/AbstractSentinelInterceptor.java

```diff
@@ -135,21 +137,43 @@ protected String getContextName(HttpServletRequest request) {
         return SENTINEL_SPRING_WEB_CONTEXT_NAME;
     }
 
+
+    /**
+     * When a handler starts an asynchronous request, the DispatcherServlet exits without invoking postHandle and afterCompletion
+     * Called instead of postHandle and afterCompletion to exit the context and clean thread-local variables when the handler is being executed concurrently.
+     * @param request the current request
+     * @param response the current response
+     * @param handler the handler (or {@link HandlerMethod}) that started async
+     * execution, for type and/or instance examination
+     */
+    @Override
+    public void afterConcurrentHandlingStarted(HttpServletRequest request, HttpServletResponse response,
+                                        Object handler) throws Exception {
+        exitContext(request);
+    }
     @Override
     public void afterCompletion(HttpServletRequest request, HttpServletResponse response,
                                 Object handler, Exception ex) throws Exception {
+        exitContext(request, ex);
+    }
+
+    private void exitContext(HttpServletRequest request) {
+        exitContext(request, null);
+    }
+
+    private void exitContext(HttpServletRequest request, Exception ex) {
```

Comment:
This function does more than just exiting the context, so naming it simply 'exit' might be more appropriate.

Reply:
ok